File: 2/P1.py

data = open("In1.txt")
rules = {"A Y":6, "A X":3, "A Z": 0, "B X": 0, 
        "B Y": 3, "B Z":6, "C X": 6, "C Y":0, "C Z":3}
shape = {"X":1, "Y":2, "Z":3}

points = 0
for line in data:
    points += rules[line[:3]] + shape[line[2]]

print(points)

File: Revisited C++/01/P1.cpp

#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main(int argc, char* argv[]) {
    string line;
    ifstream myfile;
    vector<vector<int>> values;
    myfile.open("data.txt", ios::in);
    if (myfile.is_open()){
        vector<int> current_elf = {};
        while(getline(myfile,line)){
            if(line != ""){
                current_elf.push_back(stoi(line));
            }
            else{
                values.push_back(current_elf);
                current_elf.clear();
            }
        }
        values.push_back(current_elf);
    }
    int biggest_sum = 0;
    for (auto v: values){
        int local_sum = 0;
        for (auto calorie:v){
            local_sum += calorie;
        }
        if (local_sum > biggest_sum){
            biggest_sum = local_sum;
        }
    }
    cout<<biggest_sum<<endl;
    return 0;
}